### ISSUE TITLE
DSR not working

### ISSUE THREAD

**Issue by karibertils** [cloudnativelabs/kube-router#985]
**What happened?**
I am trying to get DSR working by following the guide at https://www.kube-router.io/docs/user-guide/#direct-server-return.
I have enabled hostIPC, hostPID and mounted /var/run/docker.sock and verified it's available in the kube-router pods. I have also tried using the manifest with DSR requirements enabled.

When using kube-router with BGP and announcing external address all works fine. As soon as i annote services with "kube-router.io/service.dsr=tunnel" the external address stops responding. Removing the annotation makes the external address work again.

**What did you expect to happen?**
I expected the external address to respond and work normally

**How can we reproduce the behavior you experienced?**
1. Deploy kube-router with DSR manifest
2. Deploy nginx with external address, and verify it's working
3. Annote nginx service with DSR, and external address stops working

**Screenshots / Architecture Diagrams / Network Topologies **
The router has established BGP session for the entire time and has the external address in the routing table. Both with and without the DSR annotation.

** System Information (please complete the following information):**
 - Kube-Router Version (`kube-router --version`): Running /usr/local/bin/kube-router version v1.0.1, built on 2020-07-28T23:51:21+0000, go1.10.8
 - Kube-Router Parameters: --run-router=true --run-firewall=true --run-service-proxy=true --bgp-graceful-restart=true --kubeconfig=/var/lib/kube-router/kubeconfig --advertise-external-ip=true --advertise-cluster-ip=true --advertise-loadbalancer-ip=true --peer-router-asns=64512 --peer-router-ips=10.10.0.254 --enable-pod-egress=false --enable-overlay=false --metrics-port=10249
 [e.g. --run-router --run-service-proxy --enable-overlay --overlay-type=full etc.]
 - Kubernetes Version (`kubectl version`) : v1.19.0
 - Cloud Type: on premise
 - Kubernetes Deployment Type: kubeadm
 - Kube-Router Deployment Type: DaemonSet
 - Cluster Size: 9

** Logs, other output, metrics **
Here are logs from kube-router when i annote the nginx service
```
[kube-router-rr4hq kube-router] I0906 06:09:53.838094   22088 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-rr4hq kube-router] I0906 06:09:53.843630   22088 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-w4869 kube-router] I0906 06:09:53.843182    4783 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-4w8jr kube-router] I0906 06:09:53.845176   57227 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-bdw4g kube-router] I0906 06:09:53.847033   10584 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-rr4hq kube-router] I0906 06:09:53.851233   22088 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-w4869 kube-router] I0906 06:09:53.849794    4783 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-wl58x kube-router] I0906 06:09:53.852704    2187 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-z2bcl kube-router] I0906 06:09:53.853191   24639 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-4w8jr kube-router] I0906 06:09:53.852870   57227 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-rr4hq kube-router] E0906 06:09:53.854228   22088 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-b5lmb kube-router] I0906 06:09:53.855331    2103 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-rr4hq kube-router] I0906 06:09:53.854475   22088 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-8xmrh kube-router] I0906 06:09:53.855113    4207 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-rr4hq kube-router] E0906 06:09:53.857581   22088 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:09:53.854630   10584 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-w8qt4 kube-router] I0906 06:09:53.856355    2257 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-w4869 kube-router] I0906 06:09:53.856783    4783 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-4w8jr kube-router] I0906 06:09:53.859971   57227 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-wl58x kube-router] I0906 06:09:53.861542    2187 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-w4869 kube-router] E0906 06:09:53.860321    4783 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-b5lmb kube-router] I0906 06:09:53.863670    2103 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-z2bcl kube-router] I0906 06:09:53.864374   24639 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-w4869 kube-router] I0906 06:09:53.860666    4783 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-8xmrh kube-router] I0906 06:09:53.862349    4207 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-4w8jr kube-router] E0906 06:09:53.863999   57227 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:09:53.862780   10584 network_services_controller.go:1024] Successfully assinged VIP: 193.42.1.130 in endpoint 10.97.4.134. 
[kube-router-4w8jr kube-router] I0906 06:09:53.864372   57227 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-w8qt4 kube-router] I0906 06:09:53.864843    2257 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-w4869 kube-router] E0906 06:09:53.865072    4783 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:09:53.862909   10584 network_services_controller.go:1073] Successfully disabled rp_filter in endpoint 10.97.4.134. 
[kube-router-bdw4g kube-router] I0906 06:09:53.862928   10584 network_services_controller.go:1083] Current network namespace after revert namespace to host network namespace: NS(25: 4, 4026531992) 
[kube-router-bdw4g kube-router] I0906 06:09:53.864416   10584 network_services_controller.go:1024] Successfully assinged VIP: 213.181.99.213 in endpoint 10.97.4.134. 
[kube-router-4w8jr kube-router] E0906 06:09:53.868744   57227 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:09:53.864495   10584 network_services_controller.go:1073] Successfully disabled rp_filter in endpoint 10.97.4.134. 
[kube-router-bdw4g kube-router] I0906 06:09:53.864525   10584 network_services_controller.go:1083] Current network namespace after revert namespace to host network namespace: NS(25: 4, 4026531992) 
[kube-router-b5lmb kube-router] I0906 06:09:53.872673    2103 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-bdw4g kube-router] I0906 06:09:53.868526   10584 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-8xmrh kube-router] I0906 06:09:53.873681    4207 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-z2bcl kube-router] I0906 06:09:53.876367   24639 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-bdw4g kube-router] E0906 06:09:53.872468   10584 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:09:53.873023   10584 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-b5lmb kube-router] E0906 06:09:53.878253    2103 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-8xmrh kube-router] E0906 06:09:53.877489    4207 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] E0906 06:09:53.876593   10584 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-8xmrh kube-router] I0906 06:09:53.877872    4207 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-b5lmb kube-router] I0906 06:09:53.878954    2103 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-bdw4g kube-router] I0906 06:09:53.878085   10584 network_services_controller.go:1024] Successfully assinged VIP: 193.42.1.130 in endpoint 10.97.4.134. 
[kube-router-w8qt4 kube-router] I0906 06:09:53.878455    2257 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-z2bcl kube-router] E0906 06:09:53.883634   24639 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:09:53.878151   10584 network_services_controller.go:1073] Successfully disabled rp_filter in endpoint 10.97.4.134. 
[kube-router-8xmrh kube-router] E0906 06:09:53.881301    4207 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-b5lmb kube-router] E0906 06:09:53.883250    2103 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:09:53.878176   10584 network_services_controller.go:1083] Current network namespace after revert namespace to host network namespace: NS(25: 4, 4026531992) 
[kube-router-bdw4g kube-router] I0906 06:09:53.879793   10584 network_services_controller.go:1024] Successfully assinged VIP: 213.181.99.213 in endpoint 10.97.4.134. 
[kube-router-z2bcl kube-router] I0906 06:09:53.884127   24639 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-w8qt4 kube-router] E0906 06:09:53.882688    2257 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:09:53.879878   10584 network_services_controller.go:1073] Successfully disabled rp_filter in endpoint 10.97.4.134. 
[kube-router-w8qt4 kube-router] I0906 06:09:53.883230    2257 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-bdw4g kube-router] I0906 06:09:53.879904   10584 network_services_controller.go:1083] Current network namespace after revert namespace to host network namespace: NS(25: 4, 4026531992) 
[kube-router-wl58x kube-router] I0906 06:09:53.887892    2187 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-z2bcl kube-router] E0906 06:09:53.888718   24639 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-w8qt4 kube-router] E0906 06:09:53.887989    2257 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-wl58x kube-router] E0906 06:09:53.892235    2187 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-wl58x kube-router] I0906 06:09:53.892897    2187 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-wl58x kube-router] E0906 06:09:53.897070    2187 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-w4869 kube-router] I0906 06:09:55.712030    4783 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-w4869 kube-router] E0906 06:09:55.714001    4783 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-w4869 kube-router] I0906 06:09:55.714453    4783 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-w4869 kube-router] E0906 06:09:55.716318    4783 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-w4869 kube-router] E0906 06:09:55.724726    4783 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-w4869 kube-router] E0906 06:09:55.727854    4783 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] E0906 06:10:06.680489   10584 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] E0906 06:10:06.682957   10584 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:10:06.684430   10584 network_services_controller.go:1024] Successfully assinged VIP: 193.42.1.130 in endpoint 10.97.4.134. 
[kube-router-bdw4g kube-router] I0906 06:10:06.684501   10584 network_services_controller.go:1073] Successfully disabled rp_filter in endpoint 10.97.4.134. 
[kube-router-bdw4g kube-router] I0906 06:10:06.684527   10584 network_services_controller.go:1083] Current network namespace after revert namespace to host network namespace: NS(25: 4, 4026531992) 
[kube-router-bdw4g kube-router] I0906 06:10:06.685858   10584 network_services_controller.go:1024] Successfully assinged VIP: 213.181.99.213 in endpoint 10.97.4.134. 
[kube-router-bdw4g kube-router] I0906 06:10:06.685948   10584 network_services_controller.go:1073] Successfully disabled rp_filter in endpoint 10.97.4.134. 
[kube-router-bdw4g kube-router] I0906 06:10:06.685974   10584 network_services_controller.go:1083] Current network namespace after revert namespace to host network namespace: NS(25: 4, 4026531992) 
[kube-router-bdw4g kube-router] I0906 06:10:06.686488   10584 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-bdw4g kube-router] E0906 06:10:06.688692   10584 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:10:06.689074   10584 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-bdw4g kube-router] E0906 06:10:06.691137   10584 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-bdw4g kube-router] I0906 06:10:06.692409   10584 network_services_controller.go:1024] Successfully assinged VIP: 193.42.1.130 in endpoint 10.97.4.134. 
[kube-router-bdw4g kube-router] I0906 06:10:06.692525   10584 network_services_controller.go:1073] Successfully disabled rp_filter in endpoint 10.97.4.134. 
[kube-router-bdw4g kube-router] I0906 06:10:06.692564   10584 network_services_controller.go:1083] Current network namespace after revert namespace to host network namespace: NS(25: 4, 4026531992) 
[kube-router-bdw4g kube-router] I0906 06:10:06.693968   10584 network_services_controller.go:1024] Successfully assinged VIP: 213.181.99.213 in endpoint 10.97.4.134. 
[kube-router-bdw4g kube-router] I0906 06:10:06.694034   10584 network_services_controller.go:1073] Successfully disabled rp_filter in endpoint 10.97.4.134. 
[kube-router-bdw4g kube-router] I0906 06:10:06.694052   10584 network_services_controller.go:1083] Current network namespace after revert namespace to host network namespace: NS(25: 4, 4026531992)
```

When i remove the annotation and the service starts working again i get this in logs
```
[kube-router-b5lmb kube-router] I0906 06:24:30.704153    2103 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-b5lmb kube-router] E0906 06:24:30.706402    2103 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-b5lmb kube-router] I0906 06:24:30.706841    2103 network_services_controller.go:1762] Successfully added service: TCP:<nil>:443 (Flags: ) 
[kube-router-b5lmb kube-router] E0906 06:24:30.709162    2103 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-b5lmb kube-router] I0906 06:24:30.714820    2103 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-b5lmb kube-router] E0906 06:24:30.717156    2103 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
[kube-router-b5lmb kube-router] I0906 06:24:30.717736    2103 network_services_controller.go:1762] Successfully added service: TCP:<nil>:80 (Flags: ) 
[kube-router-b5lmb kube-router] E0906 06:24:30.720123    2103 service_endpoints_sync.go:313] Failed to delete external ip adress from dummyVipInterface due to cannot assign requested address 
```

**Additional context**
DaemonSet manifest used is here.
[kube-router-ds.yaml.txt](https://github.com/cloudnativelabs/kube-router/files/5179150/kube-router-ds.yaml.txt)

I have also tried the manifest from https://raw.githubusercontent.com/cloudnativelabs/kube-router/master/daemonset/kubeadm-kuberouter-all-features-dsr.yaml with only minor modifcation to get BGP running, with the same results.

**Comment by aauren** [comment]
@karibertils Thanks for your detailed bug report! The logs especially help narrow this down. It looks like you're facing the same error reported and fixed here: #971 

We should be releasing a 1.1 release candidate early next week that contains a fix for this. I'm going to close this for now, but if after trying kube-router from master or trying the release candidate that will get cut next week you still face this issue, please feel free to request for it to be re-opened and we'll look into it more.

**Comment by yuchunyun** [comment]
I have also encountered such a problem. Have you solved it?
